x-pack/filebeat/input/awss3/input.go

```diff
@@ -42,101 +43,124 @@ func configure(cfg *common.Config) (v2.Input, error) {
 	return newInput(config)
 }
 
-// s3Input is a input for s3
+// s3Input is a input for reading logs from S3 when triggered by an SQS message.
 type s3Input struct {
-	config config
+	config    config
+	awsConfig awssdk.Config
 }
 
 func newInput(config config) (*s3Input, error) {
-	return &s3Input{config: config}, nil
-}
-
-func (in *s3Input) Name() string { return inputName }
-
-func (in *s3Input) Test(ctx v2.TestContext) error {
-	_, err := awscommon.InitializeAWSConfig(in.config.AWSConfig)
+	awsConfig, err := awscommon.GetAWSCredentials(config.AWSConfig)
```

Comment:
`GetAWSCredentials` should be `InitializeAWSConfig` which is the new function that encapsulates the gathering creds but also enables the use of a http proxy

Reply:
Fixed. Thanks.